File: QE-Selenium-Java/Activity15.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity15 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));

        driver.get("https://training-support.net/webelements/dynamic-attributes");

        System.out.println("Page Title: " + driver.getTitle());

        // Find all input fields using dynamic attributes
        WebElement fullName = driver.findElement(
                By.xpath("//input[starts-with(@id,'full-name')]"));

        WebElement email = driver.findElement(
                By.cssSelector("input[id$='email']"));

        WebElement eventDate = driver.findElement(
                By.xpath("//input[contains(@name,'event-date')]"));

        WebElement details = driver.findElement(
                By.xpath("//textarea[contains(@id,'additional-details')]"));

        fullName.sendKeys("Devika");
        email.sendKeys("abc@xyz.com");
        eventDate.sendKeys("2026-02-15");
        details.sendKeys("Party time!");

        driver.findElement(
                By.xpath("//button[text()='Submit']")).click();

        wait.until(ExpectedConditions.visibilityOfElementLocated(
                By.id("action-confirmation")));

        String message = driver.findElement(
                By.id("action-confirmation")).getText();

        System.out.println("Message: " + message);

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 15)

driver.get("https://training-support.net/webelements/dynamic-attributes")

print("Page Title:", driver.title)

full_name = driver.find_element(
    By.XPATH, "//input[starts-with(@id,'full-name')]"
)

email = driver.find_element(
    By.CSS_SELECTOR, "input[id$='email']"
)

event_date = driver.find_element(
    By.XPATH, "//input[contains(@name,'event-date')]"
)

details = driver.find_element(
    By.XPATH, "//textarea[contains(@id,'additional-details')]"
)

full_name.send_keys("Ritesh")
email.send_keys("abc@xyz.com")
event_date.send_keys("2026-02-15")
details.send_keys("Party time!")

driver.find_element(
    By.XPATH, "//button[text()='Submit']"
).click()

wait.until(
    EC.visibility_of_element_located(
        (By.ID, "action-confirmation")
    )
)

message = driver.find_element(By.ID, "action-confirmation").text
print("Message:", message)

driver.quit()
